src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts

```diff
@@ -273,7 +275,50 @@ class CodeActionOnSaveParticipant implements ITextFileSaveParticipant {
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IInstantiationService private readonly instantiationService: IInstantiationService,
 		@ILanguageFeaturesService private readonly languageFeaturesService: ILanguageFeaturesService,
-	) { }
+		@IHostService private readonly hostService: IHostService,
+		@IEditorService private readonly editorService: IEditorService,
+		@ICodeEditorService private readonly codeEditorService: ICodeEditorService,
+	) {
+		this.hostService.onDidChangeFocus(async () => {
+			this.triggerCodeActionsCommand();
+		});
+
+		this.editorService.onDidActiveEditorChange(async () => {
+			this.triggerCodeActionsCommand();
+		});
+	}
+
+	private async triggerCodeActionsCommand() {
+		if (this.configurationService.getValue<boolean>('editor.codeActions.triggerOnFocusChange') && this.configurationService.getValue<string>('files.autoSave') === 'afterDelay') {
```

Comment:
Are there any constants we can use for these setting IDs instead?

Reply:
we don't atm